yeonjae/programmers-1/약수의합.js

```diff
@@ -0,0 +1,18 @@
+function solution(n) {
+  const arr = [];
+
+  for (let i = 1; i <= n; i++) {
+    if (n % i === 0) {
+      arr.push(i);
+    }
+  }
+
+  return arr.reduce((acc, curr) => acc + curr, 0);
+}
```

Comment:
```suggestion
function solution(n) {
  let result = 0;

  for (let i = 1; i <= n; i++) {
    if (n % i === 0) {
      result += i;
    }
  }

  return result
}
```
- push하고 또 reduce하는게 불필요한 연산같습니다! 1회 반복할 때 result에다 값을 누적시켜주고 바로 리턴하면 좋을 것 같습니다 :)

Reply:
아 그러네요!! 코멘트 감사합니다 :)